docs/api/hardhat/plugins.md

```diff
@@ -411,3 +411,108 @@ expect(await contract.getAmount()).to.equal(100);
+    zksync: true,
+    verifyURL: 'https://zksync2-testnet-explorer.zksync.dev/contract_verification'
+  }
+},
+// defaultNetwork: "zkTestnet", // optional (if not set, use '--network zkTestnet')
+```
+
+- `zkTestnet` is an arbitrary zkSync network name. You can select this as the default network using the `defaultNetwork` property.
+- `ethNetwork` is a field with the URL of the Ethereum node. You can also provide the network name (e.g. `goerli`) as the value of this field. In this case, the plugin will either use the URL of the appropriate Ethereum network configuration (from the `networks` section), or the default `ethers` provider for the network if the configuration is not provided. This field is required for all zkSync networks used by this plugin.
+- `url` is a field with the URL of the zkSync node in the case of the zkSync network (with `zksync` flag set to `true`), or the URL of the Ethereum node. This field is required for all zkSync and Ethereum networks used by this plugin.
+- `zksync` is a flag to indicate if the network represents zkSync network configuration. This field needs to be set to `true` for all zkSync networks. If you want to run a `hardhat-etherscan` verification, this field needs to be set to `false`. If set to `true`, the verification process will always try to run the verification process on the zkSync network.
+- `verifyURL` is a field that points to the verification endpoint for the specific zkSync network. This parameter is optional, and its default value is the testnet verification url (`https://zksync2-testnet-explorer.zksync.dev/contract_verification`).
+
+
+If you want to verify a smart contract on the Ethereum in the same project, it is important to add `etherscan` field in the `hardhat.config.ts` file:
+
+```typescript
+
+networks: {
+    ...
+},
+etherscan: {
+  apiKey: //<Your API key for Etherscan>,
+},
+
+```
+
+### Commands
+
+`hardhat verify --network <network> <contract address>` - verifies the contract on the given network with the given contract's address. </br>
+Note: When run like this, the verify task will try to compare compiled bytecode of all the contracts in your local setup to the deployed bytecode of the contract you are trying to verify. If there is no match, it will report an error.
+
+With the ```--contract``` parameter you can also specify which contract from your local setup you want to verify by specifying its Fully qualified name. Fully qualified name structure looks like this: "contracts/AContract.sol:TheContract" </br>
+
+Example: ``` yarn hardhat verify --network <network> <contract address> --contract <fully qualified  name>```
+
+If your contract was deployed with the specific constructor arguments, you need to specify them when running the verify task. For example: <br/>
+```yarn hardhat verify --network testnet 0x7cf08341524AAF292255F3ecD435f8EE1a910AbF "Hi there!"```
```

Comment:
having an exclamation mark at the end messed up with my validation. Not sure if it was replacing it or what, but this exact command failed with my contract

Reply:
If you are referring to the exclamation point at the end of "Hi there!" string, I put it there just as an example of a constructor parameter for a Greeter.sol contract. It doesn't have to be that specific one, and the "!" is certainly not required. Instead, you need to state that exact argument you deployed your contract with.